zest/releaser/utils.py

```diff
@@ -745,8 +745,7 @@ def _subprocess_open(p, command, input_value, show_stderr):
     if input_value:
         i.write(input_value.encode(INPUT_ENCODING))
     i.close()
-    stdout_output = o.read()
-    stderr_output = e.read()
+    (stdout_output, stderr_output) = p.communicate()
```

Comment:
This fails on all Python versions on Travis:

    ValueError: I/O operation on closed file

We could certainly be something weird and non-standard in our code. This part has been tweaked various times, trying to keep it working in various Python versions. It may not be the most modern way to do this.

Does this actually fix anything for you in practice? Or is this just a try?

Just a guess: maybe if there is an `input_value`, you should pass it to `p.communicate`. So the beginning of the function would look like this:

```
def _subprocess_open(p, command, input_value, show_stderr):
    if input_value:
        (stdout_output, stderr_output) = p.communicate(input_value.encode(INPUT_ENCODING))
    else:
        (stdout_output, stderr_output) = p.communicate()
```

Note: I have not tried this.

Reply:
I'll give this a try with what you suggested. 

It did in fact cause a problem for me when running zest.releaser in a git repo that uses git-lfs. The process actually deadlocks in that case while running 
```
DEBUG: Running command: 'git clone --depth 1 C:\Development\sources\stage gitclone'
```

With this change the deadlock is gone. I thought it may be related to what the python doc has to say about `Popen`, `PIPE`, et al (https://docs.python.org/2/library/subprocess.html):

> Note:  This will deadlock when using stdout=PIPE and/or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use communicate() to avoid that.

Reply:
For the record, the fix worked for my specific situation, but I of course forgot to run any test locally before opening the PR, sorry for that.

Reply:
So: it could be that until now it has worked fine (at least for me) only because there was not much output. Interesting.
@reinout You are using `git-lfs` something I think. Have you seen this problem?

The tests now fail with `NameError: global name 'o' is not defined`. So a bit more cleanup is needed in that function.

Reply:
I used git-annex for a while, not git-lfs.

Reply:
> The tests now fail with `NameError: global name 'o' is not defined`. So a bit more cleanup is needed in that function.

Sorry for that - I was in a hurry this morning and did (again) not test things, reasoning that "what could go wrong", and here we are :)
I'll get CI green in the course of this evening.